File: src/main/java/land/leets/domain/admin/domain/repository/AdminRepository.java
package land.leets.domain.admin.domain.repository;

import land.leets.domain.admin.domain.Admin;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;
import java.util.UUID;

public interface AdminRepository extends JpaRepository<Admin, UUID> {

    Optional<Admin> findById(String id);
}

File: src/main/kotlin/land/leets/domain/admin/domain/repository/AdminRepository.kt
package land.leets.domain.admin.domain.repository

import land.leets.domain.admin.domain.Admin
import org.springframework.data.jpa.repository.JpaRepository
import java.util.UUID

interface AdminRepository : JpaRepository<Admin, UUID> {
    fun findByUsername(username: String): Admin?
}